kraftwerk-core/src/main/java/fr/insee/kraftwerk/core/inputs/UserInputsGenesis.java

```diff
@@ -0,0 +1,149 @@
+		super(inputDirectory);
+		this.hasConfigFile = hasConfigFile;
+		this.modes=modes;
+		computeInputs();
+	}
+
+	private void computeInputs() throws KraftwerkException, IOException {
+		for (Mode mode : modes) {
+			ModeInputs modeInputs = new ModeInputs();
+			modeInputs.setDdiUrl(findDDIFile(inputDirectory.resolve(mode.name())).toFile().toURI().toURL());
+			modeInputs.setLunaticFile(findLunaticFile(inputDirectory.resolve(mode.name())));
+			modeInputs.setDataMode(mode.name());
+			if (mode == Mode.WEB || mode == Mode.TEL) {
+				modeInputs.setDataFormat("LUNATIC_XML");
+			} else {
+				// Not implemented yet
+				modeInputs.setDataFormat("OTHER");
+			}
+			if (isHasConfigFile()) {
+				modeInputs.setModeVtlFile(getModeVtlFile(mode));
+			}
+			modeInputsMap.put(mode.name(), modeInputs);
+		}
+		if (isHasConfigFile()) {
+			readUserInputs();
+		}
+	}
+
+	private Path getModeVtlFile(Mode mode) throws UnknownDataFormatException, MissingMandatoryFieldException, KraftwerkException {
+		Path userInputFile = inputDirectory.resolve(Constants.USER_INPUT_FILE);
+		try {
+			JsonNode userInputs = JsonFileReader.read(userInputFile);
+			JsonNode filesNode = userInputs.get("survey_data");
+			for (JsonNode fileNode : filesNode) {
+				String dataMode = readField(fileNode, "data_mode");
+				if (dataMode.equals(mode.name())) {
+					return convertToPath(readField(fileNode, "mode_specifications"));
+				}
+			}
+		} catch (IOException e) {
+			log.error("Unable to read user input file: {} , {}", userInputFile, e);
+			throw new UnknownDataFormatException(e.getMessage());
+		} catch (KraftwerkException e) {
+			throw e;
+		}
+		return null;
+	}
+
+	private void readUserInputs() throws UnknownDataFormatException, MissingMandatoryFieldException, KraftwerkException {
+		Path userInputFile = inputDirectory.resolve(Constants.USER_INPUT_FILE);
+		try {
+			JsonNode userInputs = JsonFileReader.read(userInputFile);
+			vtlReconciliationFile = convertToPath(readField(userInputs, "reconciliation_specifications"));
+			vtlTransformationsFile = convertToPath(readField(userInputs, "transformation_specifications"));
+			vtlInformationLevelsFile = convertToPath(readField(userInputs, "information_levels_specifications"));
+
+		} catch (IOException e) {
+			log.error("Unable to read user input file: {} , {}", userInputFile, e);
+			throw new UnknownDataFormatException(e.getMessage());
+		} catch (KraftwerkException e) {
+			throw e;
+		}
+	}
+
+	public List<String> getModes() {
+		return new ArrayList<>(modeInputsMap.keySet());
+	}
+
+	private String readField(JsonNode node, String field) throws MissingMandatoryFieldException {
+		JsonNode value = node.get(field);
+		if (value != null) {
+			String text = value.asText();
+			if (!(text.equals("") || text.equals("null"))) {
+				return text;
+			}
+		}
+		return null;
+	}
+
+	private Path convertToPath(String userField) throws KraftwerkException {
```

Comment:
convertToPath, convertToUrl in the other class = maybe a class utils

Reply:
I moved the two methods in FileUtils